### PR TITLE
Add invert_spans

### PR DESCRIPTION
**Pull request by mich11**
This adds `invert_spans(spans, parent_span)`, which returns a vector of TimeSpans that represent the gaps between an input vector of TimeSpans `spans` that are contained by the TimeSpan `parent_span`.

This function has been used in a couple of different internal projects at Beacon and was mentioned in #10 a while ago. 

WIP - I still need to add tests

### CODE REVIEW

**Review by ararslan — COMMENTED** [review]
Looks great, especially those comprehensive tests, just one suggestion to make it a bit more generic.

File: src/TimeSpans.jl [hunk 1]
```diff
@@ -360,4 +361,31 @@ Return the midpoint of a TimeSpan in `Nanosecond`s.
...
+"""
+    invert_spans(spans, parent_span)
+
+Return a vector of TimeSpans representing the gaps between a vector
+of TimeSpans `spans` that are contained within a TimeSpan
+`parent_span`.
+"""
+function invert_spans(spans::AbstractVector{TimeSpan}, parent_span::TimeSpan)
+    spans = filter(x -> contains(parent_span, x), spans)
+    spans = merge_spans((a, b) -> start(b) <= stop(a), spans)
```
[the remark this answers is not in the document]
Comment (mich11) [reply src/TimeSpans.jl:373]: Very nice 😎 

Thanks!

File: test/runtests.jl [hunk 2]
```diff
@@ -180,3 +180,44 @@ end
...
+    i_spans = invert_spans(spans, parent_span)
+    @test length(i_spans) == 7
+    @test i_spans[1] == TimeSpan(Second(0), Second(1))
+    @test all(duration.(i_spans[2:6]) .== Second(9))
+    @test i_spans[end] == TimeSpan(Second(52), stop(parent_span))
+
+    # some spans lie outside of parent_span
+    i_spans = invert_spans(spans, TimeSpan(Second(0), Second(30)))
+    @test length(i_spans) == 4
+    @test maximum(map(x -> x.stop, i_spans)) <= Second(30)
```
Comment (ararslan) [inline test/runtests.jl:207]:
```suggestion
    @test maximum(stop, i_spans) <= Second(30)
```

**Review by ararslan — APPROVED** [review]
Awesome work!

**Pull request merged**